web/pdf_viewer.js

```diff
@@ -1610,6 +1610,43 @@ class PDFViewer {
     };
   }
 
+  switchToEditAnnotationMode() {
+    const visible = this._getVisiblePages();
+    const pagesToRefresh = [];
+    const { first, ids, last, views } = visible;
+    let keepFirst = false;
+    let keepLast = false;
+    for (const page of views) {
+      const { view } = page;
+      if (!view.hasEditableAnnotations()) {
+        ids.delete(view.id);
+        continue;
+      }
+      pagesToRefresh.push(page);
+      if (page === first) {
+        keepFirst = true;
+      }
+      if (page === last) {
+        keepLast = true;
+      }
+    }
+
+    if (pagesToRefresh.length === 0) {
+      return null;
+    }
+    if (!keepFirst) {
+      visible.first = pagesToRefresh[0];
+    }
+    if (!keepLast) {
+      visible.last = pagesToRefresh.at(-1);
+    }
```

Comment:
Do we "need" the keepFirst/keepLast, i.e. how much does it save, or could we just always set the first/last pageView here and save a few lines of code overall?

Reply:
My idea was to just keep as most as possible what we had in `visible` (see https://github.com/mozilla/pdf.js/blob/e3caa3c6eeda90099dd03b451b6b4788746455b3/web/pdf_viewer.js#L1693-L1700).
If you tell me that's useless, I trust you.